scapy/contrib/automotive/doip.py

```diff
@@ -180,6 +235,25 @@ def extract_padding(self, s):
 
 
 class DoIPSocket(StreamSocket):
+    """ Custom StreamSocket for DoIP communication. This sockets automatically
+    sends a routing activation request as soon as a TCP connection is
+    established.
+
+    :param ip: IP address of destination
+    :param port: destination port, usually 13400
+    :param activate_routing: If true, routing activation request is
+                             automatically sent
+    :param source_address: DoIP source address
+    :param target_address: DoIP target address, this is automatically
+                           determined if routing activation request is sent
+    :param activation_type: This allows to set a different activation type for
+                            the routing activation request
+
+    Example:
+        >>> socket = DoIPSocket("169.254.0.131")
+        >>> pkt = DoIP(payload_type=0x8001, source_address=0xe80, target_address=0x1000) / UDS() / UDS_RDBI(identifiers=[0x1000])
+        >>> resp = socket.sr1(pkt, timeout=1)
+    """  # noqa: E501
```

Comment:
?

Reply:
Yes, `noqa` is necessary. It can also be at the end of the example code line, which actually has a length > 80, but then this `noqa` would end up in api-docs. I want to avoid that, because I think its cleaner, if no `noqa` is present in the example code. 
Also I don't wanted to break the example code line multiple times.